.github/workflows/lint-rst.yaml

```diff
@@ -0,0 +1,82 @@
+name: RST checks
+
+# Any change in triggers needs to be reflected in the concurrency group.
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - '**.rst'
+  workflow_dispatch:
+    inputs:
+      reportType:
+        description: 'rstcheck report type'
+        required: true
+        default: 'error'
+      rstfiles:
+        description: "RST files to find e.g.: -name 'README.rst' -o -name 'kind*.rst'"
+        required: true
+        default: "-name '*.rst'"
+      rstpaths:
+        description: "Paths to find e.g.: ./Documentation ./bpf"
+        required: true
+        default: "./"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.event.after }}
+  cancel-in-progress: true
+
+jobs:
+  check-rst:
+    name: Validate RST files
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          persist-credentials: false
+          fetch-depth: 0
```

Comment:
Why 0? Ideally should be '1', no?

Reply:
One '1' would be the ideal. I agree with you.

But I suppose that it won't work on a push with multiple commits where each one is related to a different `.rst`.

Even knowing the target SHA, a shallow depth clone would return `fatal: Invalid revision range before..after` after `git diff --name-only ${{ github.event.before }}..${{ github.event.after }}`.